### PR TITLE
Powershell

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Description
- Support for Powershell linting + test. Also includes some small fixes and code cleanup.

### CODE REVIEW

File: demisto_sdk/dev_tools/linter.py [hunk 1]
```diff
@@ -392,26 +397,34 @@ def _docker_image_create(self, docker_base_image, requirements):
...
+        run_params.extend([docker_image, 'sh', './{}'.format(run_script_name)])
+        print_v(f'container create: {run_params}')
+        output = subprocess.check_output(run_params, stderr=subprocess.STDOUT, universal_newlines=True)
         container_id = output.strip()
         try:
-            output = output + '\n' + run_command(' '.join(['docker', 'cp', self.project_dir + '/.', container_id +
-                                                           ':' + workdir]), exit_on_error=False)
-            output = output + '\n' + run_command(' '.join(['docker', 'cp', self.run_dev_tasks_script, container_id +
-                                                           ':' + workdir]), exit_on_error=False)
+            output = output + '\n' + subprocess.check_output(['docker', 'cp', self.project_dir + '/.', container_id +
```
Comment (yaakovi) [inline demisto_sdk/dev_tools/linter.py]: what was the problem with the run_command?

File: demisto_sdk/common/scripts/dev_scripts/pkg_dev_container_setup_pwsh.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+
+# exit on errors
+set -e
+
+echo "setting up powershell testing image..."
+
+if [ "${DEMISTO_LINT_UPDATE_CERTS}" = "yes" ]; then
+    echo "updating ca certificates ..."
+    update-ca-certificates
```
[the remark this answers is not in the document]
Comment (glicht) [reply demisto_sdk/common/scripts/dev_scripts/pkg_dev_container_setup_pwsh.sh]: It is part of openssl. Needed to add this for installing a cert inorder to install powershell stuff in docker when behind GP.

File: .flake8 [hunk 3]
```diff
@@ -7,5 +7,5 @@
 # W605 invalid escape sequence
 # E999 Invalid Syntax
 ignore = W293, W504, W291, W605, F405, F403, E999
-max-line-length = 120
+max-line-length = 160
```
[the remark this answers is not in the document]
Comment (glicht) [reply .flake8]: i think it's ok.

File: demisto_sdk/yaml_tools/unifier.py [hunk 4]
```diff
@@ -188,18 +188,21 @@ def get_data(self, extension):
...
+        :param script_type: script type: .py, .js, .ps1
         :type script_type: str
         :return: path to found code file
         :rtype: str
         """
 
         ignore_regex = (r'CommonServerPython\.py|CommonServerUserPython\.py|demistomock\.py|_test\.py'
-                        r'|conftest\.py|__init__\.py|ApiModule\.py')
+                        r'|conftest\.py|__init__\.py|ApiModule\.py'
+                        r'|CommonServerPowerShell\.ps1|CommonServerUserPowerShell\.ps1|\.Tests\.ps1')
```
[the remark this answers is not in the document]
Comment (glicht) [reply demisto_sdk/yaml_tools/unifier.py]: This is the convention of Pester. See an example at: https://github.com/pester/Pester/tree/master/Examples/Planets

File: demisto_sdk/dev_tools/linter.py [hunk 5]
```diff
@@ -392,26 +397,34 @@ def _docker_image_create(self, docker_base_image, requirements):
...
+        run_params.extend(['-e', f'PS_LINT_FILES={lint_files}'])
         if not self.root:
             run_params.extend(['-u', '{}:4000'.format(os.getuid())])
         if not self.run_args['tests']:
             run_params.extend(['-e', 'PYTEST_SKIP=1'])
+            run_params.extend(['-e', 'PS_TEST_SKIP=1'])
         if not self.run_args['pylint']:
             run_params.extend(['-e', 'PYLINT_SKIP=1'])
+        if not self.run_args['pslint']:
+            run_params.extend(['-e', 'PYLINT_SKIP=1'])
```
[the remark this answers is not in the document]
Comment (glicht) [reply demisto_sdk/dev_tools/linter.py]: It should be ps lint. will fix

File: demisto_sdk/dev_tools/linter.py (hunk 1 above)
Comment (glicht) [reply demisto_sdk/dev_tools/linter.py]: It doesn't check process exit code. also if any of the params ever have whitespace it is problematic. I kind of think we should remove it altogether and use the standard python api of `subprocess.check_output` but I didn't want to start a big refactor.

File: demisto_sdk/commands/lint/linter.py [hunk 6]
```diff
@@ -482,7 +497,19 @@ def _setup_dev_files(self, py_num):
...
+
+    def _setup_dev_files_pwsh(self):
+        # copy common server
+        try:
+            shutil.copy(self.configuration.env_dir + '/Tests/demistomock/demistomock.ps1', self.project_dir)
+            if "/Scripts/CommonServerPowerShell" not in self.project_dir:
+                # Otherwise we already have the CommonServerPowerShell.py file
+                shutil.copy(self.configuration.env_dir + '/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1',
+                            self.project_dir)
+        except Exception as e:
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/lint/linter.py]: this part should be changed in order to support base pack that should be merged today.
see:
https://github.com/demisto/demisto-sdk/pull/243/files

Comment (glicht) [reply demisto_sdk/commands/lint/linter.py]: For now let's leave the powershell common out from the base pack

**Comment by glicht** [comment]
@yuvalbenshalom base packs have been added

**Comment by glicht** [comment]
Build with powershell branch of demisto-sdk: https://github.com/demisto/content/pull/5541 is green. Merging this to master.

**Pull request merged**